src/EnergyPlus/UnitHeater.cc

```diff
@@ -1894,6 +1894,56 @@ namespace UnitHeater {
         }
     }
 
+    int GetUnitHeaterAirOutletNode(EnergyPlusData &state, int UnitHeaterNum)
+    {
+        // Return value
+        int GetUnitHeaterAirOutletNode = 0;
+
+        if (state.dataUnitHeaters->GetUnitHeaterInputFlag) {
+            GetUnitHeaterInput(state);
+            state.dataUnitHeaters->GetUnitHeaterInputFlag = false;
+        }
+
+        if (UnitHeaterNum > 0 && UnitHeaterNum <= state.dataUnitHeaters->NumOfUnitHeats) {
+            GetUnitHeaterAirOutletNode = state.dataUnitHeaters->UnitHeat(UnitHeaterNum).AirOutNode;
+        }
+
+        return GetUnitHeaterAirOutletNode;
+    }
+
+    int GetUnitHeaterAirInletNode(EnergyPlusData &state, int UnitHeaterNum)
+    {
+        int GetUnitHeaterAirInletNode = 0;
+
+        if (state.dataUnitHeaters->GetUnitHeaterInputFlag) {
+            GetUnitHeaterInput(state);
+            state.dataUnitHeaters->GetUnitHeaterInputFlag = false;
+        }
+
+        if (UnitHeaterNum > 0 && UnitHeaterNum <= state.dataUnitHeaters->NumOfUnitHeats) {
+            GetUnitHeaterAirInletNode = state.dataUnitHeaters->UnitHeat(UnitHeaterNum).AirInNode;
+        }
+
+        return GetUnitHeaterAirInletNode;
+    }
+
+    int getUnitHeaterIndex(EnergyPlusData &state, std::string_view CompName, bool &errFlag)
+    {
+        if (state.dataUnitHeaters->GetUnitHeaterInputFlag) {
+            GetUnitHeaterInput(state);
+            state.dataUnitHeaters->GetUnitHeaterInputFlag = false;
+        }
+        int EquipIndex = 0;
+        for (int UnitHeatNum = 1; UnitHeatNum <= state.dataUnitHeaters->NumOfUnitHeats; ++UnitHeatNum) {
+            if (Util::SameString(state.dataUnitHeaters->UnitHeat(UnitHeatNum).Name, CompName)) {
+                EquipIndex = UnitHeatNum;
+            }
+        }
+
+        if (EquipIndex == 0) errFlag = true;
+        return EquipIndex;
+    }
```

Comment:
I am growing concerned about the repeated getter pattern here.  Maybe it's fine.  But, I have thoughts:

- I believe this passive "if flag then call getinput and set flag to false" pattern is being moved to a more active/intentional pattern over in #10585.  Does adding all these calls make it more work to move them over there?
- Either way, as a temporary step, is it worthwhile to just move the "if flag then get input and set flag false" inside the GetInput routine itself?  It would mean a call to GetInput, but I imagine that is extremely cheap compared to the other things we are doing.  And then each of these functions could be much smaller.  Like:

```
    int getUnitHeaterIndex(EnergyPlusData &state, std::string_view CompName, bool &errFlag)
    {
        GetUnitHeaterInput(state);  // it will check and set the flag itself
        for (int UnitHeatNum = 1; UnitHeatNum <= state.dataUnitHeaters->NumOfUnitHeats; ++UnitHeatNum) {
            if (Util::SameString(state.dataUnitHeaters->UnitHeat(UnitHeatNum).Name, CompName)) {
                return UnitHeatNum;
            }
        }
        errFlag = true;
        return 0;
    }
```

I think we could also just not pass the errFlag, and instead rely on the return value from the function as a signal for error condition (zero, or negative one in the future).

I'm not saying this stuff has to be changed, I'm just surprised at the repetitive nature here.  I will also say there's a way to go much more interesting with this that leads to a broad cleanup of the objects and get inputs.  I quickly stubbed some of it out here (https://godbolt.org/z/oxPvs3jP7) for anyone interested.  Note that with the little sample link there, the array index lookup is zero/one agnostic.  The pattern would throw regardless of whether it is used with Array1D or vector.  

Anyway, I'll just carry on reviewing for now.

Reply:
@Myoldmopar Based on your suggestion. Here is a function:

    int getUnitHeaterIndex(EnergyPlusData &state, std::string\_view CompName)  
   {  
       if (state.dataUnitHeaters->GetUnitHeaterInputFlag) {  
           GetUnitHeaterInput(state);  
           state.dataUnitHeaters->GetUnitHeaterInputFlag = false;  
       }  
 

       for (int UnitHeatNum = 1; UnitHeatNum \<= state.dataUnitHeaters->NumOfUnitHeats; ++UnitHeatNum) {  
           if (Util::SameString(state.dataUnitHeaters->UnitHeat(UnitHeatNum).Name, CompName)) {  
               return UnitHeatNum;  
           }  
       }

       return 0;  
   }

Here is what I change:

1.  Remove errFlag
2.  Remove int EquipIndex
3.  Keep state.dataUnitHeaters->GetUnitHeaterInputFlag to ensure that GetUnitHeaterInput will not be called more than once

Please let me know what you think. I will make any changes after I get your feedback.

Thanks.

Reply:
That looks good, just make sure to check if the return value is zero wherever you call it and you should be good to go.  Thanks!

Reply:
@lgu1234 did you end up doing this?  It appears on GitHub like it's still as it was before...  I could be seeing it wrong.

Reply:
@Myoldmopar I completed changes last week and will upload it today.